### PR TITLE
 Fix notifications for private studies to send them only to owner

### PR DESCRIPTION
**Pull request by SlimaneAmar**
Fix notifications for private studies to send them only to owner and not to all users

### CODE REVIEW

**Review by JacquesBorsenberger — APPROVED** [review]

File: src/main/java/org/gridsuite/study/server/StudyService.java [hunk 1]
```diff
@@ -400,25 +403,23 @@ public StudyEntity doGetStudyWithPreFetchedLoadFlowResultAndUpdateIsPrivate(UUID
...
                 JsonNode node = new ObjectMapper().readTree(body).path("message");
                 if (!node.isMissingNode()) {
-                    message = node.asText();
-                    emitStudyError(studyName, UPDATE_TYPE_STUDIES, message);
+                    emitStudyCreationError(studyName, userId, isPrivate, node.asText());
                 }
             } catch (JsonProcessingException e) {
                 if (!body.isEmpty()) {
-                    emitStudyError(studyName, UPDATE_TYPE_STUDIES, body);
+                    emitStudyCreationError(studyName, userId, isPrivate, body);
```
Comment (jonenst) [inline src/main/java/org/gridsuite/study/server/StudyService.java:415]: maybe in another PR and in another repo, but I think we should not send error notifications to other users for public studies.

File: src/main/java/org/gridsuite/study/server/StudyService.java [hunk 2]
```diff
@@ -769,10 +780,12 @@ private void emitStudyChanged(UUID studyUuid, String updateType) {
         );
     }
 
-    private void emitStudyError(String studyName, String updateType, String errorMessage) {
+    private void emitStudyCreationError(String studyName, String userId, boolean isPrivate, String errorMessage) {
         studyUpdatePublisher.onNext(MessageBuilder.withPayload("")
                 .setHeader(HEADER_STUDY_NAME, studyName)
-                .setHeader(HEADER_UPDATE_TYPE, updateType)
```
[the remark this answers is not in the document]
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/study/server/StudyService.java:775]: Done

File: src/main/java/org/gridsuite/study/server/StudyService.java (hunk 1 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/study/server/StudyService.java]: Done

File: src/main/java/org/gridsuite/study/server/StudyService.java [hunk 3]
```diff
@@ -400,25 +403,23 @@ public StudyEntity doGetStudyWithPreFetchedLoadFlowResultAndUpdateIsPrivate(UUID
...
-    private Mono<? extends Throwable> handleStudyCreationError(String studyName, ClientResponse clientResponse) {
+    private Mono<? extends Throwable> handleStudyCreationError(UUID studyUuid, String studyName, String userId, ClientResponse clientResponse) {
         return clientResponse.bodyToMono(String.class).flatMap(body -> {
             try {
-                String message;
                 JsonNode node = new ObjectMapper().readTree(body).path("message");
                 if (!node.isMissingNode()) {
-                    message = node.asText();
-                    emitStudyError(studyName, UPDATE_TYPE_STUDIES, message);
+                    emitStudyCreationError(studyUuid, studyName, userId, true, node.asText());
```
Comment (jonenst) [inline src/main/java/org/gridsuite/study/server/StudyService.java:411]: if this is force to true here, then it should not be called "publicstudy", it should be called "broadcastnotification" or something like that.
But instead, I think we should send the real value of isPrivate, and add filter in the notification-server

**Review by jonenst — APPROVED** [review]

File: src/main/java/org/gridsuite/study/server/StudyService.java (hunk 3 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/study/server/StudyService.java]: Done

**Pull request merged**